### PR TITLE
Additional trace in libp2p

### PR DESCRIPTION
**Pull request by LeoHChen**
1. cherry pick EK's libp2p upgrade to s3
1. upgraded pubsub module of libp2p to the latest 0.2.5 to support trace of pubsub
1. use an environment variable to control libp2p trace support
1. increase the queue size of outbound tx message to 64 from 32

### CODE REVIEW

File: p2p/host/hostv2/hostv2.go [hunk 1]
```diff
@@ -54,16 +85,38 @@ type HostV2 struct {
...
+// It returns a nil error if and only if it has succeeded to schedule the given
+// message for sending.
+func (host *HostV2) SendMessageToGroups(groups []nodeconfig.GroupID, msg []byte) (err error) {
 	for _, group := range groups {
-		err := host.pubsub.Publish(string(group), msg)
-		if err != nil {
-			error = err
+		t, e := host.getTopic(string(group))
+		if e != nil {
+			err = e
```
Comment (rlan35) [inline p2p/host/hostv2/hostv2.go]: Should log each of the error separately right?

Comment (LeoHChen) [reply p2p/host/hostv2/hostv2.go]: the error will be surface to the caller, which is sendMesagewithRetry/WithoutRetry, we will still see the error from the top level caller.

**Comment by Daniel-VDM** [comment]
Jenkins test this please

**Comment by Daniel-VDM** [comment]
jenkins test this please

**Comment by Daniel-VDM** [comment]
jenkins test this please

**Comment by Daniel-VDM** [comment]
jenkins test this please

**Comment by Daniel-VDM** [comment]
jenkins test this please

**Comment by LeoHChen** [comment]
To clarify, I have tested it on my foundational nodes, can join network and consensus.

**Pull request merged**